pkg/agent/runtime.go

```diff
@@ -102,8 +103,26 @@ var DockerContainerRuntimeImplementsContainerRuntime ContainerRuntime = (*Docker
 func (c *DockerContainerRuntime) PullImage(ctx context.Context,
 	image string, skipTLS bool, authStr string,
 	cfg RunConfig) error {
-	authBytes := base64.URLEncoding.EncodeToString([]byte(authStr))
-	out, err := c.client.ImagePull(ctx, image, types.ImagePullOptions{RegistryAuth: string(authBytes)})
+	authBytes := []byte(authStr)
```

Comment:
Looks like we can reduce the scope of `authBytes` to the `if authStr != ""` condition

Reply:
Fixed.